fix(linter): skip extending config files that look like named configs or not files

camchenry opened the pull request:
- fixes https://github.com/oxc-project/oxc/issues/9825

For now, we are skipping any paths that look like they are a named config that starts with `eslint:` or `plugin:`. Otherwise, we also heuristically check if the path contains a dot, just to see if it even looks like a file with an extension at all.

camchenry commented:
* **#9932** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9932?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9932?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9932).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Mar 20, 11:42 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
fix(linter): skip extending config files that look like named configs or not files (#9932)

- fixes https://github.com/oxc-project/oxc/issues/9825

For now, we are skipping any paths that look like they are a named config that starts with `eslint:` or `plugin:`. Otherwise, we also heuristically check if the path contains a dot, just to see if it even looks like a file with an extension at all.
